source/extensions/filters/http/ratelimit/config.cc

```diff
@@ -28,12 +28,20 @@ Http::FilterFactoryCb RateLimitFilterConfig::createFilterFactoryFromProtoTyped(
   const uint32_t timeout_ms = PROTOBUF_GET_MS_OR_DEFAULT(proto_config, timeout, 20);
   Filters::Common::RateLimit::ClientFactoryPtr client_factory =
       Filters::Common::RateLimit::rateLimitClientFactory(context);
-  return [client_factory, timeout_ms,
+  // if ratelimit service config is provided in both bootstrap and filter, we should validate that
+  // they are same.
+  if (proto_config.has_rate_limit_service() && client_factory->rateLimitConfig().has_value() &&
+      !Envoy::Protobuf::util::MessageDifferencer::Equals(*client_factory->rateLimitConfig(),
+                                                         proto_config.rate_limit_service())) {
+    throw EnvoyException("rate limit service config in filter does not match with bootstrap");
```

Comment:
Instead of throwing an exception, should the filter config just override the bootstrap config? That may be helpful for users doing migrations part-by-part.

Reply:
I am not sure if silently overriding the bootstrap configuration with the one in filter is the right thing to do here. Also we discussed earlier to throw an exception if they do not match.